lib/backend/backend.go

```diff
@@ -78,3 +83,21 @@ type Config struct {
 	// Tablename where to store K/V in DynamoDB
 	Tablename string `yaml:"table_name,omitempty"`
 }
+
+// Config is used for 'storage' config section. It's a combination of
+// values for various backends: 'boltdb', 'etcd', 'filesystem' and 'dynamodb'
+type Config struct {
+	// Type can be "bolt" or "etcd" or "dynamodb"
+	Type string `yaml:"type,omitempty"`
+
+	EtcdConfig   `yaml:",inline"`
```

Comment:
what if there are overlapping fields? There would be collision. I would like to avoid that.